tests/auth_tests/templates/context_processors/auth_attrs_perms.html

```diff
@@ -1,4 +1,4 @@
 {% if perms.auth %}Has auth permissions{% endif %}
 {% if perms.auth.add_permission %}Has auth.add_permission permissions{% endif %}
 {% if perms.nonexistent %}nonexistent perm found{% endif %}
-{% if perms.auth.nonexistent in perms %}auth.nonexistent perm found{% endif %}
+{% if 'perms.auth.nonexistent' in perms %}auth.nonexistent perm found{% endif %}
```

Comment:
I don't think this is correct. As I read the commit: 

d5a4f209c3889a76a23a19f3212ab2d7b5c62e1c added three lines to this file `auth_attrs_perms.html`, all of them without quotes, checking that the previous behaviour (of accessing the context dict values directly) continued to work, and extending the coverage there to more examples (rather than just `perms.auth`). 

It then also added `auth_attrs_perm_in_perms.html `, plus a view for that, and an additional test case, which checked the new behaviour of string access, so `'auth.nonexisting'` and so on.

It's the added `auth_attrs_perm_in_perms.html ` that tests the new behaviour.

Reply:
I see you're point that the quotes shouldn't be here. Thanks.

I changed this to check the variable is defined before use. I could also add a new test that shows that without the variable check, an exception is raised, just let me know.

If a variable is undefined, an `in` check shouldn't implicitly result in `False`. That could result in leaking private data. For example:

```
{% if thing.undefined in public_values %}
   <p>Public stuff</p>
{% else %}
   <p>Private stuff</p>
{% endif %}
```

The safest approach is to raise an exception.